atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -51,7 +51,7 @@ public String getShortTableNameFromMappingTable(
             return unmappingCache.get(fullTableName, () -> {
                 SqlConnection conn = null;
                 try {
-                    conn = connectionSupplier.getNewUnsharedConnection();
+                    conn = connectionSupplier.get();
```

Comment:
Curious to know how you detected the deadlock and what exactly is the deadlock in having Oracle getting newConnections and closing them. Its a cache so we dont expect too many new connections.